fl4health/preprocessing/autoencoders/vae_training.py

```diff
@@ -0,0 +1,51 @@
+from typing import Tuple
+
+import numpy as np
+import torch
+
+
+class AETransformer:
+    """Transform function to replace the target in the dataset with data for
+    self-supervised approaches like VAEs. For a conditional model, it can concatenate
+    the condition with the data sample.
+    """
+
+    def __init__(self, condition: str = "", img_dims: int = 1) -> None:
+        """Initializes the AETransformer with an optional condition.
+
+        Args:
+            condition (str, optional): Condition for the transformation. Defaults to "".
```

Comment:
The `img_dims` argument isn't documented in this doc-string. The name sort of suggestions it is related to image data, but the rest of the transformer seems to be more general than that?